src/Native/System.Native/pal_io.cpp

```diff
@@ -443,3 +464,93 @@ int32_t MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     return mkstemps(pathTemplate, suffixLength);
 }
+
+static
+void ConvertPalPollFdToNative(PollFd& pal, pollfd& native)
+{
+    native.fd = pal.FileDescriptor;
+    native.events = pal.EventsToPollFor;
+    native.revents = pal.ReceivedEvents;
+}
+
+static
+void ConvertNativePollfdToPal(pollfd& native, PollFd& pal)
+{
+    pal.FileDescriptor = native.fd;
+    pal.EventsToPollFor = native.events;
+    pal.ReceivedEvents = native.revents;
+}
+
+extern "C"
+int32_t Poll(PollFd* pollData, uint32_t numberOfPollFds, int32_t timeout)
+{
+    // Convert our standardized pollfd to the native one
+    pollfd* fds = new pollfd[numberOfPollFds];
```

Comment:
I'd prefer to keep it this way; making it stackalloc'd with a limit seems really brittle and any future callers will need to be aware of that fact. By keeping it like the underlying function, all future callers are protected and can call it with as many pollfd's as they would like without having to go look / potentially change the PAL layer going forward. If you don't think we'll have many more callers in the future and are okay with that tradeoff, I can go ahead and make it stackalloc'd instead :)